### PR TITLE
Vg 75 cart improvements

### PR DESCRIPTION
**Pull request by Danback**
What was added in this PR:

animation when adding something to cart.

Cart shows how many products it has inside it.

Add scroll function to cart dropdown.

Adding more than one of the same product into the cart now makes  the item stack.

Cart dropdown closes when pressing anywhere else on the page

### CODE REVIEW

**Review by marcusju23 — CHANGES_REQUESTED** [review]
Would be nice if items in cart was smaller but other than that, lgtm 👍

**Comment by Danback** [comment]
@marcusju23 Look at latest push, should look better now

File: src/components/Navbar.vue [hunk 1]
```diff
@@ -36,85 +35,97 @@
...
+        </button>
+
+        <div v-if="isCartOpen" ref="cartDropdown" class="z-50 absolute right-[-45%] top-14 w-[400px] text-white backdrop-blur-lg bg-black/30">
+          <div class="max-h-[300px] overflow-y-auto">
+            <ul v-for="(cartProduct, index) in cartStore.products" :key="index">
+              <li>
+                <div class="flex items-center">
+                  <img class="p-2 max-h-16" :src="cartProduct.image" :alt="cartProduct.title">
+                  <div class="p-2 grow">
+                    <p class="text-lg">{{ cartProduct.title }}</p>
```
Comment (marcusju23) [inline src/components/Navbar.vue]: Maybe `text-sm` instead? 🤔

**Review by marcusju23 — CHANGES_REQUESTED** [review]
Minor thing that could make more items fit, what do you think? 🤔

File: src/components/Navbar.vue (hunk 1 above)
Comment (marcusju23) [reply src/components/Navbar.vue]: I'm also getting a lot of warnings in the console, is it similar for you? 
image

Comment (Danback) [reply src/components/Navbar.vue]: Do we want more items to fit without scrolling? I can add that I guess. As for the warnings I'm uncertain, I'll look into it.

Comment (Danback) [reply src/components/Navbar.vue]: I think I know what's causing the warnings, I'll try to fix

**Comment by Danback** [comment]
@marcusju23 Fixed the requests, no more warnings as well

**Review by marcusju23 — APPROVED** [review]

**Review by Rishad11 — APPROVED** [review]
Well done, it looks cleaner now!

**Pull request merged**